commit cabef4d095dec75e85eb4ae7e5f124179ff8fd44 (refs/changes/86/4186/4)
Author: Rui Loura <rui@joyent.com>
Date:   2018-06-27T20:33:38+00:00 (1 year, 3 months ago)
    
    MANTA-3753 muppet should leverage autometadata instead of checking nictags
    Reviewed by: Jared Morrow <jared.morrow@joyent.com>
    Approved by: Cody Mello <cody.mello@joyent.com>

diff --git a/muppet.js b/muppet.js
index f7f8290..6b00ba8 100644
--- a/muppet.js
+++ b/muppet.js
@@ -58,29 +58,35 @@ function getUntrustedIPs(cfg, callback) {
 
         cfg.log.info({ nics: nics }, 'Looked up NICs');
 
-        function addIPsFromNics(nic) {
-            // Skip NICs on trusted networks.
-            if (nic.nic_tag === 'admin' || nic.nic_tag === 'manta') {
+        function _pushIP(ip) {
+            /* If this is an admin, manta, or other trusted IP, skip it. */
+            if ((cfg.adminIPS && cfg.adminIPS.indexOf(ip) !== -1) ||
+                (cfg.mantaIPS && cfg.mantaIPS.indexOf(ip) !== -1) ||
+                ip === cfg.trustedIP)  {
+
                 return;
             }
 
+            if (!net.isIPv4(ip) && !net.isIPv6(ip)) {
+                return;
+            }
+
+            cfg.untrustedIPs.push(ip);
+        }
+
+        function _addIPsFromNics(nic) {
             if (nic.hasOwnProperty('ips')) {
                 nic.ips.forEach(function parseIP(addr) {
-                    const ip = addr.split('/')[0];
-                    if (net.isIPv4(ip) || net.isIPv6(ip)) {
-                        cfg.untrustedIPs.push(ip);
-                    }
+                    _pushIP(addr.split('/')[0]);
                 });
             } else if (nic.hasOwnProperty('ip')) {
-                if (net.isIPv4(nic.ip)) {
-                    cfg.untrustedIPs.push(nic.ip);
-                }
+                _pushIP(nic.ip);
             } else {
                 cfg.log.warn({ nic: nic }, 'NIC has no IP addresses');
             }
         }
 
-        nics.forEach(addIPsFromNics);
+        nics.forEach(_addIPsFromNics);
         callback();
     });
 }
@@ -135,6 +141,12 @@ function configure() {
     try {
         const _f = opts.file || __dirname + '/etc/config.json';
         cfg = JSON.parse(fs.readFileSync(_f, 'utf8'));
+        if (cfg.adminIPS && typeof (cfg.adminIPS) === 'string') {
+            cfg.adminIPS = cfg.adminIPS.split(',');
+        }
+        if (cfg.mantaIPS && typeof (cfg.mantaIPS) === 'string') {
+            cfg.mantaIPS = cfg.mantaIPS.split(',');
+        }
     } catch (e) {
         log.fatal(e, 'unable to parse %s', _f);
         process.exit(1);
diff --git a/sapi_manifests/muppet/template b/sapi_manifests/muppet/template
index 17c3478..6a16306 100644
--- a/sapi_manifests/muppet/template
+++ b/sapi_manifests/muppet/template
@@ -1,6 +1,8 @@
 {
   "name": "manta.{{REGION}}.{{DNS_DOMAIN}}",
   "trustedIP": "{{auto.MANTA_IP}}",
+  "adminIPS":  "{{auto.ADMIN_IPS}}",
+  "mantaIPS":  "{{auto.MANTA_IPS}}",
   "zookeeper": {
     "servers": [
       {{#ZK_SERVERS}}
